### ISSUE TITLE
Propsd should support local property overrides

### ISSUE THREAD

**Issue by dgreene-r7** [rapid7/propsd#226]
We should allow override layers that take the highest precedent that can be pointed to in the config file. I envision it like so:

`/etc/propsd/config.json`
```
{
  "index": {
    "bucket": "my-bucket"
  },
  "properties": {
    "metadata": "some piece of metadata"
  },
  "overrides": [
    "/etc/propsd/overrides/*"
  ]
}
```

`/etc/propsd/overrides/metadata.json`
```
{
  "metadata": "this overrides the metadata defined in the config"
}
```

The `overrides` array should support glob syntax. 

The override files should be JSON and will be loaded as a `Layer.Dynamic` (or something similar) with a file watcher on it that reloads the layer on change.

**Comment by fmitchell-r7** [comment]
I'd be hesitant to allow glob syntax without a strict definition of how files matching the glob are ordered. Suppose I have three files, 10.json, 1.json, and 2.json. Are they ordered 1.json, 10.json, 2.json or 1.json, 2.json, 10.json?

**Comment by dgreene-r7** [comment]
Good call. Maybe we make a check to see if we're using a glob pattern and require a comparator callback?

**Comment by erran-r7** [comment]
> I'd be hesitant to allow glob syntax without a strict definition of how files matching the glob are ordered.

>  Maybe we make a check to see if we're using a glob pattern and require a comparator callback?

I'm not sure we need to make it any stricter or more compicated — we *should* document how the glob strings behave: Do they behave like bash or is the glob expansion different due to using nodejs?

```
bash-3.2$ echo *
1.json 10.json 2.json
```
